pkg/kubectl/cmd/create_job.go

```diff
@@ -37,110 +34,96 @@ var (
 		Create a job with the specified name.`))
 
 	jobExample = templates.Examples(i18n.T(`
+		# Create a job
+		kubectl create job my-job --image=busybox
+
 		# Create a job from a CronJob named "a-cronjob"
 		kubectl create job --from=cronjob/a-cronjob`))
 )
 
-type CreateJobOptions struct {
-	Name string
-	From string
-
-	Namespace string
-	Client    clientbatchv1.BatchV1Interface
-	Out       io.Writer
-	DryRun    bool
-	Builder   *resource.Builder
-	Cmd       *cobra.Command
-}
-
-// NewCmdCreateJob is a command to ease creating Jobs from CronJobs.
-func NewCmdCreateJob(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
-	c := &CreateJobOptions{
-		Out: cmdOut,
-	}
+// NewCmdCreateJob is a command to create a new job.
+func NewCmdCreateJob(f cmdutil.Factory, cmdOut, cmdErr io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "job NAME [--from-cronjob=CRONJOB]",
+		Use:     "job NAME [--from=cronjob/name --image=image] -- [COMMAND] [args...]",
 		Short:   jobLong,
 		Long:    jobLong,
 		Example: jobExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(c.Complete(f, cmd, args))
-			cmdutil.CheckErr(c.RunCreateJob())
+			argsLenAtDash := cmd.ArgsLenAtDash()
+			cmdutil.CheckErr(createJob(f, cmdOut, cmdErr, cmd, args, argsLenAtDash))
 		},
 	}
 	cmdutil.AddApplyAnnotationFlags(cmd)
 	cmdutil.AddValidateFlags(cmd)
 	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, cmdutil.JobV1GeneratorName)
+	cmd.Flags().String("image", "", "Image name to run.")
 	cmd.Flags().String("from", "", "The name of the resource to create a Job from (only cronjob is supported).")
-
 	return cmd
 }
 
-func (c *CreateJobOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) (err error) {
-	if len(args) == 0 {
+func createJob(f cmdutil.Factory, cmdOut, cmdErr io.Writer, cmd *cobra.Command, args []string, argsLenAtDash int) error {
+	if len(args) == 0 || argsLenAtDash == 0 {
 		return cmdutil.UsageErrorf(cmd, "NAME is required")
 	}
-	c.Name = args[0]
-
-	c.From = cmdutil.GetFlagString(cmd, "from")
-	c.Namespace, _, err = f.DefaultNamespace()
-	if err != nil {
-		return err
+	name := args[0]
+	image := cmdutil.GetFlagString(cmd, "image")
+	from := cmdutil.GetFlagString(cmd, "from")
+	if len(image) == 0 && len(from) == 0 {
+		return fmt.Errorf("image or from must be specified")
 	}
-
-	clientset, err := f.KubernetesClientSet()
-	if err != nil {
-		return err
+	if len(image) != 0 && len(from) != 0 {
+		return fmt.Errorf("image or from must be specified, not both")
```

Comment:
nit: might be worth adding `--` in front of image and from to make it clear you are referring to flag names